File: Problems/0001_0099/0002_Add_Two_Number/Project_Scala/OperateListNode.scala
object OperateListNode {
    def createListNode(flds:String):ListNode = {
        if (flds.length <= 0)
            return null
        return createSubListNode(Mylib.stringToIntArray(flds), 0)
    }

    def createSubListNode(nums:Array[Int], index:Int):ListNode = {
        if (index >= nums.size)
            return null
        
        var node:ListNode = new ListNode(nums(index))
        node.next = createSubListNode(nums, index + 1)

        return node
    }

    def listNodeToString(ll:ListNode):String = {
        if (ll == null)
            return ""

        var retStr:String = ll.x.toString

        if (ll.next != null)
            retStr += " -> " + listNodeToString(ll.next)

        return retStr
    }
}

File: Problems/0001_0099/0002_Add_Two_Number/Project_Python3/ListNode/OperateListNode.py
from .ListNode import ListNode

class OperateListNode:
    def createListNode(self, flds):
        if len(flds) <= 0:
            return None
        nums = [int(val) for val in flds.split(",")]
        return self.createSubListNode(nums, 0)

    def createSubListNode(self, nums, index):
        if index >= len(nums):
            return None

        node = ListNode(nums[index])
        node.next = self.createSubListNode(nums, index + 1)

        return node

    def ListNodeToString(self, ll):
        if ll is None:
            return ""
        retStr = str(ll.val) 

        if ll.next is not None:
            retStr += " -> " + self.ListNodeToString(ll.next)
        return retStr
